lib/vector-vrl-functions/src/get_metadata_field.rs

```diff
@@ -19,15 +26,15 @@ impl Function for GetMetadataField {
     fn parameters(&self) -> &'static [Parameter] {
         &[Parameter {
             keyword: "key",
-            kind: kind::BYTES,
+            kind: kind::ANY,
             required: true,
         }]
     }
 
     fn examples(&self) -> &'static [Example] {
         &[Example {
             title: "Get the datadog api key",
-            source: r#"get_metadata_field("datadog_api_key")"#,
+            source: r#"get_metadata_field(.datadog_api_key)"#,
```

Comment:
Do we want to change the example to push people to using the `secret` functions?